kubernetes: build kubelet from source & add openssl

ijc opened the pull request:
This builds the binaries for the kubernetes image from scratch rather than pulling unverified images of the Internet.

I also added openssl to the container since I found I needed it to sign my csr for RBAC purposes.

I had hoped to also bump cri-containerd here, but need to sort out a new build-time dependency on `libapparmor-dev` (which is only in Alpine edge/testing at the moment) so that will come later.

rn commented on projects/kubernetes/kubernetes/Dockerfile:
```diff
@@ -28,15 +56,19 @@ RUN apk add --no-cache --initdb -p /out \
+RUN cp _output/bin/kubeadm /out/usr/bin/kubeadm
+RUN cp _output/bin/kubectl /out/usr/bin/kubectl
+
+RUN strip /out/usr/bin/kubelet /out/usr/bin/kubeadm /out/usr/bin/kubectl
```
You also strip above, once should be enough

ijc replied:
Dang, I _knew_ I did it on Friday but missed it when I checked so I added it again this morning, Doh!

justincormack replied:
I don't believe it is safe to use `strip` on Go binaries. You can use the Go build options that we use in our build script that does a partial `strip`.

ijc replied:
What are the symptoms? These ones worked well enough to run the sock shop demo, but perhaps the issues are subtle?

justincormack commented on projects/kubernetes/kubernetes/Dockerfile:
```diff
@@ -1,14 +1,42 @@
 ENV kubernetes_version v1.7.6
 ENV cni_version        v0.6.0
 
 ENV kube_release_artefacts "https://dl.k8s.io/${kubernetes_version}/bin/linux/amd64"
```
this is not being used?

ijc commented:
Force pushed with:
* Dropped both strips for now until the reasons for avoid are understood (the kube build system needs a bit more investigation if we are to add `-ldflags "-s"` which I think is what @justincormack was referring to).
* Dropped the now unused env
* New patch to disable cAdvisor on public port (the default in kubelet).

The apparmor issues with cri-containerd are resolved but that is still not updated here due to a new issue which is a requirement for `/dev/disk/by-uuid` which I am investigating.
